### PR TITLE
Onboarding monitor & applicationinsights for autogeneration

### PR DESCRIPTION
**Pull request by leni-msft**
1. monitor/resource-manager
```
> azure-schema-generator@1.0.0 generate-single /home/nilei/workspace/azure-resource-manager-schemas/generator
> ts-node cmd/generatesingle "monitor/resource-manager"

[/tmp/schm_azspc] executing: git fsck
[/tmp/schm_azspc] executing: git reset -q .
[/tmp/schm_azspc] executing: git checkout -q -- .
[/tmp/schm_azspc] executing: git clean -q -fd
[/tmp/schm_azspc] executing: git gc
[/tmp/schm_azspc] executing: git fetch -q
[/tmp/schm_azspc] executing: git checkout -q origin/master
Using whitelist config:
{
  "basePath": "monitor/resource-manager",
  "namespace": "Microsoft.Insights",
  "resourceConfig": [
    {
      "type": "diagnosticSettings",
      "scopes": 28
    },
    {
      "type": "diagnosticSettingsAssociation",
      "scopes": 28
    },
    {
      "type": "guestDiagnosticSettingsAssociation",
      "scopes": 28
    }
  ]
}
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/den0h8g7ajs --tag=all-api-versions --api-version=2017-03-01-preview --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[27.32 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2017-03-01-preview/Microsoft.Insights.json
Provider Namespace: Microsoft.Insights
API Version: 2017-03-01-preview
Resource Types (Resource Group Scope):
- activityLogAlerts
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/d62j7nc4pnv --tag=all-api-versions --api-version=2017-05-01-preview --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[20.45 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2017-05-01-preview/Microsoft.Insights.json
Provider Namespace: Microsoft.Insights
API Version: 2017-05-01-preview
Resource Types (Management Group Scope):
- diagnosticSettings
Resource Types (Subscription Scope):
- diagnosticSettings
Resource Types (Resource Group Scope):
- diagnosticSettings
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/ev0dx9uwnnv --tag=all-api-versions --api-version=2017-11-01-preview --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[23.54 s] Generation Complete
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/kuglhtzvizs --tag=all-api-versions --api-version=2017-12-01-preview --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[17.95 s] Generation Complete
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/71y35w3skr9 --tag=all-api-versions --api-version=2018-06-01-preview --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[21.21 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2018-06-01-preview/Microsoft.Insights.json
Provider Namespace: Microsoft.Insights
API Version: 2018-06-01-preview
Resource Types (Management Group Scope):
- guestDiagnosticSettingsAssociation
Resource Types (Subscription Scope):
- guestDiagnosticSettingsAssociation
Resource Types (Resource Group Scope):
- guestDiagnosticSettings
- guestDiagnosticSettingsAssociation
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/h0la7s9tnl --tag=all-api-versions --api-version=2018-11-27-preview --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[18.8 s] Generation Complete
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/jgcc9fp92zd --tag=all-api-versions --api-version=2019-10-17-preview --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[25.69 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2019-10-17-preview/Microsoft.Insights.json
Provider Namespace: Microsoft.Insights
API Version: 2019-10-17-preview
Resource Types (Resource Group Scope):
- privateLinkScopes
- privateLinkScopes/privateEndpointConnections
- privateLinkScopes/scopedResources
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/e0h9bpd41qi --tag=all-api-versions --api-version=2015-04-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[31.61 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2015-04-01/Microsoft.Insights.json
Provider Namespace: Microsoft.Insights
API Version: 2015-04-01
Resource Types (Resource Group Scope):
- autoscalesettings
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/ldh7nztcst --tag=all-api-versions --api-version=2015-07-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[28.19 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2015-07-01/Microsoft.Insights.json
Provider Namespace: Microsoft.Insights
API Version: 2015-07-01
Resource Types (Management Group Scope):
- diagnosticSettings
Resource Types (Subscription Scope):
- diagnosticSettings
Resource Types (Resource Group Scope):
- diagnosticSettings
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/sikyflbop1 --tag=all-api-versions --api-version=2016-03-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[31.48 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2016-03-01/Microsoft.Insights.json
Provider Namespace: Microsoft.Insights
API Version: 2016-03-01
Resource Types (Subscription Scope):
- logprofiles
Resource Types (Resource Group Scope):
- alertrules
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/h0uz2aljyi --tag=all-api-versions --api-version=2016-09-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[29.01 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2016-09-01/Microsoft.Insights.json
Provider Namespace: Microsoft.Insights
API Version: 2016-09-01
Resource Types (Management Group Scope):
- diagnosticSettings
Resource Types (Subscription Scope):
- diagnosticSettings
Resource Types (Resource Group Scope):
- diagnosticSettings
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/mmu0oiauthd --tag=all-api-versions --api-version=2017-04-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[29.97 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2017-04-01/Microsoft.Insights.json
Provider Namespace: Microsoft.Insights
API Version: 2017-04-01
Resource Types (Resource Group Scope):
- actionGroups
- activityLogAlerts
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/7f1jxh6sq2 --tag=all-api-versions --api-version=2018-01-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[43.29 s] Generation Complete
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/9gyjcw0fp1k --tag=all-api-versions --api-version=2018-03-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[32.89 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2018-03-01/Microsoft.Insights.json
Provider Namespace: Microsoft.Insights
API Version: 2018-03-01
Resource Types (Resource Group Scope):
- actionGroups
- metricAlerts
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/9cnojl202l --tag=all-api-versions --api-version=2018-04-16 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[27.67 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2018-04-16/Microsoft.Insights.json
Provider Namespace: Microsoft.Insights
API Version: 2018-04-16
Resource Types (Resource Group Scope):
- scheduledQueryRules
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/1afg9uz9wp8 --tag=all-api-versions --api-version=2018-09-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[25.85 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2018-09-01/Microsoft.Insights.json
Provider Namespace: Microsoft.Insights
API Version: 2018-09-01
Resource Types (Resource Group Scope):
- actionGroups
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/54v7ifn8qfg --tag=all-api-versions --api-version=2019-03-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[23.57 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2019-03-01/Microsoft.Insights.json
Provider Namespace: Microsoft.Insights
API Version: 2019-03-01
Resource Types (Resource Group Scope):
- actionGroups
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/fm4nbyh5ahg --tag=all-api-versions --api-version=2019-06-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/monitor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[22.65 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2019-06-01/Microsoft.Insights.json
Provider Namespace: Microsoft.Insights
API Version: 2019-06-01
Resource Types (Resource Group Scope):
- actionGroups
================================================================================================================================
```
2. applicationinsights/resource-manager
```
> azure-schema-generator@1.0.0 generate-single /home/nilei/workspace/azure-resource-manager-schemas/generator
> ts-node cmd/generatesingle "applicationinsights/resource-manager"

[/tmp/schm_azspc] executing: git fsck
[/tmp/schm_azspc] executing: git reset -q .
[/tmp/schm_azspc] executing: git checkout -q -- .
[/tmp/schm_azspc] executing: git clean -q -fd
[/tmp/schm_azspc] executing: git gc
[/tmp/schm_azspc] executing: git fetch -q
[/tmp/schm_azspc] executing: git checkout -q origin/master
Using whitelist config:
{
  "basePath": "applicationinsights/resource-manager",
  "namespace": "Microsoft.Insights",
  "suffix": "ApplicationInsights",
  "resourceConfig": [
    {
      "type": "workbooks",
      "scopes": 8
    },
    {
      "type": "workbooktemplates",
      "scopes": 8
    }
  ]
}
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/o0bkroaa9z --tag=all-api-versions --api-version=2017-10-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/applicationinsights/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[14.58 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2017-10-01/Microsoft.Insights.ApplicationInsights.json
Provider Namespace: Microsoft.Insights
API Version: 2017-10-01
Resource Types (Resource Group Scope):
- components/pricingPlans
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/6clsw3ipnqt --tag=all-api-versions --api-version=2018-05-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/applicationinsights/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[12.87 s] Generation Complete
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/9jfv64tklj5 --tag=all-api-versions --api-version=2018-06-17-preview --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/applicationinsights/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[14.71 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2018-06-17-preview/Microsoft.Insights.ApplicationInsights.json
Provider Namespace: Microsoft.Insights
API Version: 2018-06-17-preview
Resource Types (Resource Group Scope):
- workbooks
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/p5znvtvjqf --tag=all-api-versions --api-version=2019-09-01-preview --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/applicationinsights/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[12.28 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2019-09-01-preview/Microsoft.Insights.ApplicationInsights.json
Provider Namespace: Microsoft.Insights
API Version: 2019-09-01-preview
Resource Types (Resource Group Scope):
- queryPacks
- queryPacks/queries
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/s1d11zrditn --tag=all-api-versions --api-version=2019-10-17-preview --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/applicationinsights/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[11.81 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2019-10-17-preview/Microsoft.Insights.ApplicationInsights.json
Provider Namespace: Microsoft.Insights
API Version: 2019-10-17-preview
Resource Types (Resource Group Scope):
- workbooktemplates
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/axqks5z9xz --tag=all-api-versions --api-version=2020-02-02-preview --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/applicationinsights/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[11.77 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2020-02-02-preview/Microsoft.Insights.ApplicationInsights.json
Provider Namespace: Microsoft.Insights
API Version: 2020-02-02-preview
Resource Types (Resource Group Scope):
- components
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/6ou1yplyth5 --tag=all-api-versions --api-version=2020-03-01-preview --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/applicationinsights/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[11.45 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2020-03-01-preview/Microsoft.Insights.ApplicationInsights.json
Provider Namespace: Microsoft.Insights
API Version: 2020-03-01-preview
Resource Types (Resource Group Scope):
- components/linkedStorageAccounts
================================================================================================================================
[/home/nilei/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/lneqhbsljpp --tag=all-api-versions --api-version=2015-05-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/applicationinsights/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.12.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/nilei/.autorest/@autorest_core@3.0.6257/node_modules/@autorest/core/dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[17.14 s] Generation Complete
================================================================================================================================
Filename: /home/nilei/workspace/azure-resource-manager-schemas/schemas/2015-05-01/Microsoft.Insights.ApplicationInsights.json
Provider Namespace: Microsoft.Insights
API Version: 2015-05-01
Resource Types (Resource Group Scope):
- components
- components/analyticsItems
- components/Annotations
- components/currentbillingfeatures
- components/exportconfiguration
- components/favorites
- components/myanalyticsItems
- components/ProactiveDetectionConfigs
- webtests
- workbooks
================================================================================================================================
```

### CODE REVIEW

**Comment by leni-msft** [comment]
@anthony-c-martin please note that commit 27d9b0996aadce5dfac3bcb019fa3c0965d5c082, since the subscriptionScope/managementGroupScope root schema structure has been changed.

File: generator/constants.ts [hunk 1]
```diff
@@ -17,15 +17,15 @@ export const resourceGroupRootSchema = {
 };
 export const subscriptionRootSchema = {
     file: path.join(schemasBasePath, '2018-05-01/subscriptionDeploymentTemplate.json'),
-    jsonPath: 'properties.resources.items.oneOf[0].allOf[1].oneOf'
```
Comment (anthony-c-martin) [inline generator/constants.ts]: Thanks for fixing this!

**Comment by anthony-c-martin** [comment]
@ifeoluwaokunoren please could you pull someone in from the relevant team to review?

**Comment by MarcusFelling** [comment]
@ifeoluwaokunoren would you be able to take a look at this for @anthony-c-martin?

**Pull request closed without merging**